std/conv.d

```diff
@@ -2273,25 +2295,36 @@ to an integral value.
 Params:
     Target = the integral type to convert to
     s = the lvalue of an input range
+    doCount = the flag for deciding to report the number of consumed characters
 
 Returns:
-    A number of type `Target`
+    A number of type `Target` if doCount is set to No.doCount
+    A `tuple` containing a number of type `Target` and a `size_t` if doCount is set to Yes.doCount
 
 Throws:
     A $(LREF ConvException) If an overflow occurred during conversion or
     if no character of the input was meaningfully converted.
 */
-Target parse(Target, Source)(ref Source s)
+auto parse(Target, Source, Flag!"doCount" doCount = No.doCount)(ref Source s)
 if (isSomeChar!(ElementType!Source) &&
     isIntegral!Target && !is(Target == enum))
 {
     static if (Target.sizeof < int.sizeof)
     {
         // smaller types are handled like integers
-        auto v = .parse!(Select!(Target.min < 0, int, uint))(s);
-        auto result = ()@trusted{ return cast(Target) v; }();
-        if (result == v)
-            return result;
+        auto v = .parse!(Select!(Target.min < 0, int, uint), Source, Yes.doCount)(s);
+        auto result = ()@trusted{ return cast(Target) v[0]; }();
+        if (result == v[0])
+        {
+            static if (doCount)
+            {
+                return tuple(result, v[1]);
+            }
+            else
+            {
+                return result;
```

Comment:
`result == v[0]` already so this is equivalent to `return v;` 
You might want to rename `result` to `truncate` and write this as 
```d
auto truncate = ()@trusted{ return cast(Target) v[0]; }();
if (v[0] == truncate)
    return v;
```

Reply:
My understanding is that the difference between result and v[0] is their type, which is why I was explicitly returning result. I tried the modification and it breaks both pre-existing and newly-added tests (I was returning v and v[0] respectively).

Reply:
Hmm. Maybe 
```d
if (v[0] == result)
{
    static if (doCount) return v;
    else return result;
}
```

Reply:
Sorry for the radio silence, I tried this and it's still failing. It fails because the expected return value is the same type as result, not v[0].

Reply:
@thewilsonator I'm guessing it can't be done as you're suggesting as it seems that the cast is important. With your suggestion the types will differ.